fetcher-v2/utils/elector-node.go

```diff
@@ -0,0 +1,145 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+)
+
+const INTERVAL = time.Second * 10
+
+type ElectorNode struct {
+	Host      host.Host
+	Ps        *pubsub.PubSub
+	Topic     *pubsub.Topic
+	Sub       *pubsub.Subscription
+	Data      []string
+	NodeMutex sync.Mutex
+	Cancel    context.CancelFunc
+}
+
+func NewElectorNode(host host.Host, ps *pubsub.PubSub, topicString string) (*ElectorNode, error) {
+	topic, err := ps.Join(topicString)
+	if err != nil {
+		return nil, err
+	}
+
+	sub, err := topic.Subscribe()
+	if err != nil {
+		return nil, err
+	}
+
+	return &ElectorNode{
+		Host:  host,
+		Ps:    ps,
+		Topic: topic,
+		Sub:   sub,
+		Data:  make([]string, 0),
+	}, nil
+}
+
+func (n *ElectorNode) nodeReadiness(rt pubsub.PubSubRouter, topic string) (bool, error) {
+	if topic == n.Topic.String() {
+		return true, nil
+	}
+	return false, fmt.Errorf("not enough peers ready")
+}
+
+func (n *ElectorNode) Start() {
+	ctx, cancel := context.WithCancel(context.Background())
+	n.Cancel = cancel
+
+	go n.subscribe(ctx)
+	go n.publish(ctx)
+}
+
+func (n *ElectorNode) subscribe(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			//log.Println("Stopping health checker node")
```

Comment:
I would recommend using `log.Debug` and set appropriate log level to avoid commenting or uncommenting log code.